sched/clock/clock_getcpuclockid.c

```diff
@@ -0,0 +1,74 @@
+/****************************************************************************
+ * sched/clock/clock_getcpuclockid.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <time.h>
+#include <errno.h>
+#include <unistd.h>
+
+#include "clock/clock.h"
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: clock_getcpuclockid
+ *
+ * Description:
+ *   The function shall return clock id of the CPU-time clock of the process
+ *   specified by pid
+ *
+ * Input Parameters:
+ *   pid - the specified process id
+ *   clockid - the clock type that need to setup
+ *
+ * Returned Value:
+ *   Return 0 on success, return error number on error
+ *
+ ****************************************************************************/
+
+int clock_getcpuclockid(pid_t pid, FAR clockid_t *clockid)
+{
+  if (pid < 0)
+    {
+      set_errno(EINVAL);
+      return ERROR;
+    }
+
+  /* If the pid is 0, we need to use the pid of current process */
+
+  if (pid == 0)
+    {
+      pid = getpid();
+    }
+
+  /* For clock_getcpuclockid, the clock type are
+   * CLOCK_PROCESS_CPUTIME_ID
+   */
+
+  *clockid = (pid << CLOCK_SHIFT) | CLOCK_PROCESS_CPUTIME_ID;
+  return OK;
```

Comment:
Can this be a part of libc?

Reply:
do you mean that this "clock_getcpuclockid()" method should not defined as a syscall, but as a libc method direct ?

Reply:
Yes. I do not see a reason for adding new system call.
We can use `pid = _SCHED_GETPID();` and implement it as a libc function

Reply:
`clock_getres` can be easily moved to libc as well while `clock_gettime` should remain a sys call. @xiaoxiang781216 what is your opinion on this? I think that simple functions that do not need interaction with kernel services should be placed in libc to reduce the number of system calls.

Reply:
I'm wrong with the `clock_getres` assumption. It should reside in the kernel. We can proceed with the current approach and optimize later. Please fix C89 incompatibility and let's move forward.